### PR TITLE
fix: Show Twitter Area as Collapsible/Extendable Area 

### LINKED ISSUE #6121: Mobile Screens: Show Twitter Area as Collapsible/Extendable Area
On mobile pages the Twitter area is diverting the attention of the user a bit. On the other hand the Twitter area potentially provides important updates for attendees of an event. Moving the Twitter area to the bottom of the page is not suitable as there are areas which load on scrolling. The best solution for now seems to be therefore as follows:
* change the Twitter area into a collapsible/extendable area with a triangle arrow in front. Show it like that: `▶ Recent Tweets`
* make the area collapsed by default
* when expanded show a down arrow: `▼ Recent Tweets`

Screenshot from 2020-12-23 10-51-56

### LINKED ISSUE #6122: Mobile Views: Make Event Dropdown Menu Stay on Top When user Scrolls Down Further
Many pages have this feature where a special menu stays on top of the page when a user scrolls down. Please implement the same thing for the drop-down event menu on public event pages.

When the user scrolls down keep the menu on top of the screen. When the user reaches the area "Tickets" show the menu item "Tickets" in the menu and so forth. Implement this on all public event pages for the mobile view.

Screenshot from 2020-12-23 11-03-47

### PR DESCRIPTION
**Pull request by ranjitkshah**
Fixes #6121 

#### Short description of what this resolves:
self-explanatory

####screenshot:
https://www.loom.com/share/c3d3bf9aad704706945a7fb64310b6fe

-
-
-

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Review by MeghalBisht — CHANGES_REQUESTED** [review]
please correct the indentation!

File: app/templates/components/widgets/twitter-timeline.hbs [hunk 1]
```diff
@@ -1,7 +1,50 @@
...
+      {{handle}}
+    </a>
+    <script
+      async
+      src="https://platform.twitter.com/widgets.js"
+      charset="utf-8"
+    ></script>
+  {{/unless}}
+  {{#if this.device.isMobile}}
+    <UiAccordion>
```
Comment (daretobedifferent18) [inline app/templates/components/widgets/twitter-timeline.hbs:22]: there is a code duplication pls improve the code

Comment (daretobedifferent18) [reply app/templates/components/widgets/twitter-timeline.hbs:22]: ke changes here instead make changes where the componenet is being used.

**Comment by daretobedifferent18** [comment]
@ranjitkshah Pls make changes as asked above and I think test fail will also be fixed after that

File: app/templates/components/widgets/twitter-timeline.hbs (hunk 1 above)
Comment (ranjitkshah) [reply app/templates/components/widgets/twitter-timeline.hbs]: > @ranjitkshah Pls make changes as asked above and I think test fail will also be fixed after that

changes done pls check

**Comment by ranjitkshah** [comment]
> @ranjitkshah Pls make changes as asked above and I think test fail will also be fixed after that

changes done pls check

File: app/templates/components/widgets/twitter-timeline.hbs [hunk 2]
```diff
@@ -4,4 +4,4 @@
     {{t 'Tweets by'}} @{{handle}}
   </a>
   <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
-{{/if}}
+{{/if}}
```
Comment (daretobedifferent18) [inline app/templates/components/widgets/twitter-timeline.hbs:7]: add line at end of file

**Review by daretobedifferent18 — CHANGES_REQUESTED** [review]
Current | Expected 
-------- | ---------
Screenshot_2020-12-23 Codeheat Ask Me Anything - January Edition eventyay | Screenshot from 2020-12-23 17-23-23

currently recent tweets is looking like a simple text. 
pls change as shown above

**Comment by ranjitkshah** [comment]
> Current	Expected
> Screenshot_2020-12-23 Codeheat Ask Me Anything - January Edition eventyay	Screenshot from 2020-12-23 17-23-23
> currently recent tweets is looking like a simple text.
> pls change as shown above

done

**Review by daretobedifferent18 — APPROVED** [review]
Working fine for me

**Review by MeghalBisht — COMMENTED** [review]
Please add dropdown icon inside!

**Comment by ranjitkshah** [comment]
done 
image

**Comment by ranjitkshah** [comment]
done
image

**Review by daretobedifferent18 — APPROVED** [review]
Great @ranjitkshah

**Review by MeghalBisht — DISMISSED** [review]

File: app/templates/public.hbs [hunk 3]
```diff
@@ -50,7 +50,17 @@
           {{#if (and this.device.isMobile this.twitterLink)}}
               <div class="item">
                 <div class="twitter content">
-                  <Widgets::TwitterTimeline @handleOrProfile={{this.twitterLink.link}} />
+                  <UiAccordion>
+                    <div class="title" >
+                      <span class="ui button pl-2">
+                      <i class="dropdown icon"></i>
+                        Recent tweets
+                      </span>
```
Comment (daretobedifferent18) [inline app/templates/public.hbs:58]:
```suggestion
                      <span class="ui button pl-2">
                        <i class="dropdown icon"></i>
                        Recent tweets
                      </span>
```

Comment (daretobedifferent18) [reply app/templates/public.hbs:58]: Pls do the following changes

Comment (ranjitkshah) [reply app/templates/public.hbs]: done

**Review by daretobedifferent18 — DISMISSED** [review]

**Comment by maze-runnar** [comment]
would it be ok if all buttons left aligned, recent tweets position seems odd.
Screenshot from 2020-12-24 00-50-38

**Comment by mariobehling** [comment]
Screenshot_2020-12-23 Codeheat Ask Me Anything - January Edition eventyay

The UI becomes too cluttered currently. Therefore please revert back to this version without gray area.

However, make the space between the arrow and the text > Recent tweets the same space as between the clock icon and the date "Thursday..".

**Comment by ranjitkshah** [comment]
> Screenshot_2020-12-23 Codeheat Ask Me Anything - January Edition eventyay
> 
> The UI becomes too cluttered currently. Therefore please revert back to this version without gray area.
> 
> However, make the space between the arrow and the text > Recent tweets the same space as between the clock icon and the date "Thursday..".

I think it looks good now, Pls check
image

**Comment by mariobehling** [comment]
Thanks. It is good, but it seems it is not exactly pixel correct as there are slight differences in the spaces here and there. The > icon seems to have a bit more space than the clock icon and the blue boxes are not exactly in line with the gray box on their right side. I wish you would see this yourself.

Screenshot from 2020-12-24 13-20-04
Screenshot from 2020-12-24 13-22-49

**Comment by ranjitkshah** [comment]
@mariobehling one more clarification, I did not change the alignment of the blue buttons So, all three should be in one line or only blue buttons & recent tweet in one line

**Review by daretobedifferent18 — COMMENTED** [review]
PLs make all the buttons to align at same distance from left in mobile view

**Comment by ranjitkshah** [comment]
change:
image

File: app/styles/partials/utils.scss [hunk 4]
```diff
@@ -73,14 +73,18 @@
   padding-right: .25rem !important;
 }
 
-.pr-2 {
```
Comment (iamareebjamal) [inline app/styles/partials/utils.scss:76]: Why removing this?

Comment (ranjitkshah) [reply app/styles/partials/utils.scss:76]: sorry, got confused with pl-2, pl-2 is unused css

**Review by maze-runnar — APPROVED** [review]

**Pull request merged**